// Copyright (C) 2018 The Android Open Source Project
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto2";

package emulator_location;

// Protobuf definition of the metadata for
// a single point of Location data.

message PointMetadata {
  // Name that the user gave this point
  required string logical_name = 1;

  // Creation time
  required int64 creation_time = 2;

  // Latitude degrees [-90.0 .. +90.0]
  required double latitude = 3;

  // Longitude degrees [-180.0 .. +180.0]
  required double longitude = 4;

  // Altitude meters
  required double altitude = 5;

  // Descriptive string given by the user
  optional string description = 6;

  // Street address
  optional string address = 7;
}
